Mark `ShootSystemComponentsHealthy` condition unhealthy if `ManagedResource` is stuck in progressing for more than threshold time (#7241)
gardener/gardener: docs/concepts/gardenlet.md

- `EveryNodyReady`: The conditions of the worker nodes are checked (e.g., `Ready`, `MemoryPressure`, etc.). Also, it's checked whether the Kubernetes version of the installed `kubelet` matches the desired version specified in the `Shoot` resource.
- `SystemComponentsHealthy`: The conditions of the `ManagedResource`s are checked (e.g. `ResourcesApplied`, etc.). Also, it is verified whether the VPN tunnel connection is established (which is required for the `kube-apiserver` to communicate with the worker nodes).

> Sometimes there can be case where ManagedReosurce get stuck with both `Healthy` and `Progressing` condition set to `true`. To prevent it one can set `GardenletConfiguration.controllers.shootCare.managedResourceProgressingThreshold`. So if ManagedReosurce `Progressing` condition is `true` for more than `managedResourceProgressingThreshold`, `SystemComponentsHealthy` condition is marked `Progressing` or `False` accordingly.

Reviewer: ```suggestion
Sometimes, `ManagedResource`s can have both `Healthy` and `Progressing` conditions set to `True` (e.g., when a `DaemonSet` rolls out one-by-one on a large cluster with many nodes) while this is not reflected in the `Shoot` status. In order to catch issues where the rollout gets stuck, one can set `.controllers.shootCare.managedResourceProgressingThreshold` in the `gardenlet`'s component configuration. If the `Progressing` condition is still `True` for more than the configured duration, the `SystemComponentsHealthy` condition in the `Shoot` is set to `False`, eventually.
```

> Sometimes there can be case where ManagedReosurce get stuck with both `Healthy` and `Progressing` condition set to `true`. To prevent it one can set `GardenletConfiguration.controllers.shootCare.managedResourceProgressingThreshold`. So if ManagedReosurce `Progressing` condition is `true` for more than `managedResourceProgressingThreshold`, `SystemComponentsHealthy` condition is marked `Progressing` or `False` accordingly.
> 

Reviewer 2: ```suggestion
If `GardenletConfiguration.controllers.shootCare.managedResourceProgressingThreshold` is set and the ManagedResource is in `Progressing=True` state for more than `managedResourceProgressingThreshold`, `SystemComponentsHealthy` condition is set to `False`.
```
